src/ChangePrecision.jl

```diff
@@ -85,7 +85,9 @@ function changeprecision(T, x::Float64)
     end
 end
 function changeprecision(T, ex::Expr)
-    if Meta.isexpr(ex, :call, 3) && ex.args[1] == :^ && ex.args[3] isa Int
+    if Meta.isexpr(ex, :macrocall)
+        return changeprecision(T, macroexpand(ChangePrecision, ex))
```

Comment:
This isn't right — if anything, we would want to expand in the caller's module.

Reply:
So just `macroexpand(@__MODULE__, ex)`?

Reply:
Note that the caller's module is passed to the `@changeprecision` macro as `__module__`, so from there it can be passed to the `changeprecision` function.

Reply:
So it has to be passed downstream to every `changeprecision` method? Should `include` also do this?

Reply:
Yes, it has to be passed downstream through every method.

Reply:
See last commit

Reply:
Upon reflection, I'm not sure this will be the right thing, e.g. if the user evaluates a module definiton.